python-backend/app/api/mines/status/resources/status.py

```diff
@@ -1,18 +1,29 @@
 from flask_restplus import Resource
+from werkzeug.exceptions import BadRequest, NotFound
+
 from ..models.mine_status import MineStatus
+from app.api.mines.status.models.mine_status_xref import MineStatusXref
 from app.extensions import api
 from ....utils.access_decorators import requires_role_mine_view
 from ....utils.resources_mixins import UserMixin, ErrorMixin
-from ....constants import MINE_STATUS_OPTIONS
+from app.api.mines.mine_api_models import MINE_STATUS_CODE_MODEL
+
+
+class MineStatusListResource(Resource, UserMixin):
+    @api.marshal_with(MINE_STATUS_CODE_MODEL, envelope='records')
+    @requires_role_mine_view
+    def get(self):
+        return MineStatusXref.active_status_codes()
 
 
 class MineStatusResource(Resource, UserMixin, ErrorMixin):
     @api.doc(params={'mine_status_guid': 'Mine status guid.'})
     @requires_role_mine_view
-    def get(self, mine_status_guid=None):
-        if mine_status_guid:
-            mine_status = MineStatus.find_by_mine_status_guid(mine_status_guid)
-            return mine_status.json() if mine_status else self.create_error_payload(
-                404, 'Mine Status not found'), 404
-        else:
-            return {'options': MINE_STATUS_OPTIONS}
+    def get(self, mine_status_guid):
+        if not mine_status_guid:
+            raise BadRequest('A mine guid must be provided.')
```

Comment:
Does MineStatusResource have multiple routes to this GET? If not, this method will never be called without a `mine_status_guid` url param

Reply:
Yes, there is one route to the MineStatusResource which requires a mine_status_guid and one route to the list resource which has no path params.

Reply:
Yeah, but if they're different resources then this `get` will always have the `mine_status_guid`, no? Otherwise, it'll go to the other `get` method?

Reply:
True, the error catch is unnecessary, I will remove it.

Reply:
Cool, thanks. I just took a look, and `find_by_mine_status_guid` doesn't check that `mine_status_guid` is a valid guid, so replacing the presence check with a uuid validation probably makes the most sense 😃 I can't remember if an invalid uuid on a SELECT query causes a runtime error